Fix the `fmt` compiler warning

Using `g++ 13.3.0`, I get the following compiler warning when compiling `libsemigroups`:

```
In file included from ./third_party/fmt-11.1.4/include/fmt/format.h:41,
                 from ./third_party/fmt-11.1.4/include/fmt/chrono.h:23,
                 from [...]/libsemigroups/include/libsemigroups/detail/fmt.hpp:40,
                 from [...]/libsemigroups/include/libsemigroups/exception.hpp:26,
                 from [...]/libsemigroups/include/libsemigroups/dot.hpp:30,
                 from [...]/libsemigroups/include/libsemigroups/forest.hpp:31,
                 from src/forest.cpp:19:
In constructor 'constexpr fmt::v11::detail::value<Context>::value(unsigned int) [with Context = fmt::v11::context]',
    inlined from 'std::string fmt::v11::format(format_string<T ...>, T&& ...) [with T = {unsigned int&, unsigned int, unsigned int&, long unsigned int&}]' at ./third_party/fmt-11.1.4/include/fmt/format.h:4200:49,
    inlined from 'void libsemigroups::Forest::throw_if_not_acyclic(node_type, node_type) const' at src/forest.cpp:154:30:
./third_party/fmt-11.1.4/include/fmt/base.h:2132:56: warning: 'p' may be used uninitialized [-Wmaybe-uninitialized]
 2132 |   constexpr FMT_INLINE value(unsigned x FMT_BUILTIN) : uint_value(x) {}
      |                                                        ^~~~~~~~~~~~~
src/forest.cpp: In member function 'void libsemigroups::Forest::throw_if_not_acyclic(node_type, node_type) const':
src/forest.cpp:139:17: note: 'p' was declared here
  139 |       node_type p;
      |                 ^
At global scope:
cc1plus: note: unrecognized command-line option '-Wno-unused-command-line-argument' may have been intended to silence earlier diagnostics
```

--- a/src/forest.cpp
+++ b/src/forest.cpp
@@ -135,8 +135,8 @@ namespace libsemigroups {
     size_t              current = 0;
 
     for (Forest::node_type m = 0; m != number_of_nodes(); ++m) {
-      auto      n = m;
-      node_type p;
+      auto      n      = m;
+      node_type p      = UNDEFINED;
       size_t    length = 0;
       while (n != UNDEFINED && seen[n] == not_yet_seen) {
         seen[n] = current;

